src/eterna/EPars.ts

```diff
@@ -364,12 +378,26 @@ export default class EPars {
             return null;
         }
 
-        let index: number = parenthesis.indexOf('(.)');
+        let index = parenthesis.indexOf('()');
+        if (index === -1) index = parenthesis.indexOf('[]');
+        if (index === -1) index = parenthesis.indexOf('{}');
+        if (index === -1) index = parenthesis.indexOf('<>');
+        if (index >= 0) {
+            return `There is a length 0 hairpin loop which is impossible at base ${index + 2}`;
+        }
+
+        index = parenthesis.indexOf('(.)');
```

Comment:
Commenting just so we have a record: 
This does not account for loops of the form `([)`. Will discuss with dev/lab members how we ought to handle this in the future.